src/solvers/flattening/boolbv_constant.cpp

```diff
@@ -81,14 +81,6 @@ bvt boolbvt::convert_constant(const constant_exprt &expr)
   {
     const auto &value = expr.get_value();
 
-    if(value.size() != width)
-    {
-      error().source_location=expr.find_source_location();
-      error() << "wrong value length in constant: "
-              << expr.pretty() << eom;
-      throw 0;
-    }
-
```

Comment:
I don't immediately see why ```value.size()``` is guaranteed to be ```width```, could this become an invariant?

Reply:
It's not guaranteed. The change removes the check.